packages/protocol/contracts/stability/Exchange.sol

```diff
@@ -141,56 +216,33 @@ contract Exchange is
     }
 
     emit Exchanged(msg.sender, sellAmount, buyAmount, sellGold);
-    return buyAmount;
   }
 
   /**
-   * @dev Returns the amount of buyToken a user would get for sellAmount of sellToken
+   * @notice Returns the amount of buy tokens a user would get for sellAmount of the sell token
    * @param sellAmount The amount of sellToken the user is selling to the exchange
    * @param sellGold `true` if gold is the sell token
    * @return The corresponding buyToken amount.
    */
   function getBuyTokenAmount(uint256 sellAmount, bool sellGold) external view returns (uint256) {
     if (sellAmount == 0) return 0;
-    uint256 sellTokenBucket;
-    uint256 buyTokenBucket;
-    (buyTokenBucket, sellTokenBucket) = getBuyAndSellBuckets(sellGold);
 
-    FixidityLib.Fraction memory reducedSellAmount = getReducedSellAmount(sellAmount);
-    FixidityLib.Fraction memory numerator = reducedSellAmount.multiply(
-      FixidityLib.newFixed(buyTokenBucket)
-    );
-    FixidityLib.Fraction memory denominator = FixidityLib.newFixed(sellTokenBucket).add(
-      reducedSellAmount
-    );
-
-    // Can't use FixidityLib.divide because denominator can easily be greater
-    // than maxFixedDivisor.
-    // Fortunately, we expect an integer result, so integer division gives us as
-    // much precision as we could hope for.
-    return numerator.unwrap().div(denominator.unwrap());
+    (uint256 buyTokenBucket, uint256 sellTokenBucket) = getBuyAndSellBuckets(sellGold);
```

Comment:
Should we put the short circuit above (return 0) in the internal version of the function?